Nodejs/Product/Analysis/Analysis/Analyzer/AnalysisUnit.cs

```diff
@@ -188,9 +188,9 @@ internal AnalysisUnit(JsAst ast, EnvironmentRecord environment)
         }
 
         internal virtual void AnalyzeWorker(DDG ddg, CancellationToken cancel) {
-            if (Tree != ProjectEntry.Tree) {
-                // we were enqueued and a new version became available, don't re-analyze against
-                // the old version.
+            if (Ast != null && ProjectEntry != null && ProjectEntry.Tree != null && Tree != ProjectEntry.Tree) {
```

Comment:
Should these ever be null? Is it safe to fall through to the remaining code if they are?